### ISSUE TITLE
io.js tests fail on windows

### ISSUE THREAD

**Issue by springmeyer** [mapbox/node-pre-gyp#147]
With io.js the node-pre-gyp tests that shell out to `npm test` for each test app fail on windows/appveyor with `Module did not self-register.`

https://ci.appveyor.com/project/Mapbox/node-pre-gyp/build/1.0.216/job/2psngm8nbqjl9w7g#L373

No idea yet what is going wrong.

**Comment by springmeyer** [comment]
disabled in https://github.com/mapbox/node-pre-gyp/commit/312448615f8ca962f73126f15deef11f13c92c9b just to keep appveyor green for node.js until this mystery is solved.

**Comment by Mithgol** [comment]
I am almost sure that's iojs/io.js#751.

**Comment by springmeyer** [comment]
@Mithgol - thank you - that certainly looks like the problem.

**Comment by Mithgol** [comment]
On iojs/io.js#751 they're going to fix this problem on the engine's level, but it can also be fixed on a build tools' level — TooTallNate/node-gyp#564.

(That's a good thing unless some [bystander effect](https://en.wikipedia.org/wiki/Bystander_effect) happens.)

**Comment by springmeyer** [comment]
Okay. But I'm thinking this is not a major problem overall. I think this is only an issue for how the node-pre-gyp tests work and I can adapt (working on this now). The critical `lib/testbinary.js` test I think is working for io.js because it actually uses `process.execPath` will brings in the right binary name. 

```sh
$ node -e "console.log(process.execPath)"
/Users/dane/.nvm/versions/io.js/v1.2.0/bin/iojs
```

So, to recap: this should not prevent node-pre-gyp working for users of io.js at all. Right?

**Comment by Mithgol** [comment]
Right.

(Starting from https://github.com/TooTallNate/node-gyp/pull/564#issuecomment-70396795 some comments seem to imply that `process.execPath` might become less reliable in the future. However, that's merely a distant possibility.)

**Comment by springmeyer** [comment]
fixed in #148
